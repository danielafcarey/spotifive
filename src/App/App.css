.Header {
  text-align: right;
  padding-right: 15px; }
  .Header h1 {
    display: inline-block;
    font-size: 55px;
    margin: 0;
    padding: 10px;
    vertical-align: middle; }
  .Header img {
    display: inline-block;
    width: 65px;
    padding: 30px 6px;
    vertical-align: middle; }
  .Header .profile-photo {
    position: absolute;
    left: 10px;
    border-radius: 50%;
    object-fit: cover; }

.search {
  width: 45%;
  margin: 50px auto;
  text-align: center; }
  .search input {
    width: 90%;
    height: 45px;
    font-size: 20px;
    box-sizing: border-box;
    border: none;
    padding-left: 10px;
    margin: 30px 0 0 0; }
    .search input::placeholder {
      padding-left: 10px;
      color: grey; }
  .search p {
    text-align: left; }
  .search .search-error {
    font-size: 16px;
    padding: 10px 0 0 35px;
    margin-top: 0; }

.Instructions {
  padding-top: 30px; }
  .Instructions .instruction {
    display: flex;
    flex-direction: row;
    align-items: center;
    padding: 0 20px; }
    .Instructions .instruction p {
      padding: 0 15px; }

.search-results ul {
  padding: 0 20px; }

.search-results p {
  text-align: center; }

.artist-card {
  display: flex;
  flex-direction: row;
  align-items: center; }
  .artist-card:hover {
    cursor: pointer;
    background-color: #272727; }
  .artist-card img {
    height: 50px;
    width: 50px;
    padding: 10px;
    border-radius: 50%;
    object-fit: cover; }
  .artist-card p {
    text-align: left;
    padding: 0 15px;
    font-size: 25px; }

.signin {
  width: 45%;
  margin: 50px auto;
  text-align: center; }

.tryit {
  width: 45%;
  margin: 50px auto;
  text-align: center; }
  .tryit .artist-image {
    width: 300px;
    height: 300px;
    border-radius: 50%;
    object-fit: cover; }
  .tryit p {
    padding-top: 0; }
  .tryit ul {
    list-style: none;
    text-align: left; }
  .tryit .track {
    display: flex;
    flex-direction: row;
    align-items: center; }
    .tryit .track li {
      font-size: 16px;
      padding: 10px; }

.success {
  width: 45%;
  margin: 50px auto;
  text-align: center; }
  .success button {
    margin-top: 30px; }
  .success .artist-image {
    width: 300px;
    height: 300px;
    border-radius: 50%;
    object-fit: cover; }
  .success p {
    padding-top: 0; }
  .success ul {
    list-style: none;
    text-align: left; }
  .success .track {
    display: flex;
    flex-direction: row;
    align-items: center; }
    .success .track li {
      font-size: 16px;
      padding: 10px; }

body {
  background: #333333;
  background: -moz-linear-gradient(top, #333333 0%, #000000 40%);
  background: -webkit-linear-gradient(top, #333333 0%, #000000 40%);
  background: linear-gradient(to bottom, #333333 0%, #000000 50%);
  filter: progid:DXImageTransform.Microsoft.gradient( startColorstr=$hexColor1, endColorstr=$hexColor2,GradientType=0 );
  height: 150vh;
  margin: 0;
  padding: 0;
  font-family: 'Roboto', sans-serif;
  color: white; }
  body button {
    background-color: #1fd760;
    width: 90%;
    height: 45px;
    border: none;
    color: white;
    font-size: 20px; }
    body button:hover {
      cursor: pointer;
      background-color: #1ab350; }
  body p {
    padding-top: 30px;
    font-size: 30px;
    font-weight: 300; }

@media only screen and (max-width: 768px) {
  .signin,
  .search,
  .tryit,
  .success {
    width: 75%; }
  .Header {
    padding-right: 20px; }
    .Header h1 {
      font-size: 35px; }
    .Header img {
      width: 35px;
      padding: 30px 0; }
    .Header .profile-photo {
      left: 20px; }
  body p {
    font-size: 20px;
    padding-bottom: 10px; }
  .signin {
    margin-top: 0; }
  .search,
  .tryit,
  .success {
    margin-top: 10px; }
  .instruction img {
    width: 30px; }
  .artist-card p {
    font-size: 20px;
    padding: 6px;
    margin: 0; }
  .artist-card img {
    width: 35px;
    height: 35px; }
  .tryit .artist-image,
  .success .artist-image {
    width: 150px;
    height: 150px; }
  .tryit ul,
  .success ul {
    padding-left: 20px; } }

@media only screen and (max-width: 400px) {
  body button {
    font-size: 16px; }
  .search .search-error {
    padding-left: 15px; } }
