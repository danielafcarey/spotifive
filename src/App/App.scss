@import '../Header/Header.scss';
@import '../Authorize/Authorize.scss';
@import '../Search/Search.scss';
@import '../Instructions/Instructions.scss';
@import '../SearchResults/SearchResults.scss';
@import '../ArtistCard/ArtistCard.scss';
@import '../Signin/Signin.scss';
@import '../Tryit/Tryit.scss';
@import '../Success/Success.scss';

$spotify-green: #1fd760;

@mixin linear-gradient($hexColor1, $hexColor2) {
  background: $hexColor1;
  background: -moz-linear-gradient(top, $hexColor1 0%, $hexColor2 50%);
  background: -webkit-linear-gradient(top, $hexColor1 0%, $hexColor2 50%);
  background: linear-gradient(to bottom, $hexColor1 0%, $hexColor2 50%);
  filter: progid:DXImageTransform.Microsoft.gradient( startColorstr=$hexColor1, endColorstr=$hexColor2,GradientType=0 );
}

body {
  @include linear-gradient(#333333, #000000);
  height: 100vh;
  margin: 0;
  padding: 0;
  font-family: 'Roboto', sans-serif;
  color: white;

  button {
    background-color: $spotify-green;
    width: 90%;
    height: 45px;
    border: none;
    color: white;
    font-size: 20px;

    &:hover {
      cursor: pointer;
      background-color: darken($spotify-green, 8%);
    }
  }

  p {
    padding-top: 30px;
    font-size: 30px;
    font-weight: 300;
  }
}


@media only screen and (max-width: 768px) {
  .signin,
  .search,
  .tryit,
  .success {
    width: 75%
  } 

  .Header {
    text-align: center;

    h1 {
      font-size: 40px;
    }

    img {
      width: 45px;
    }
  }

  body p {
    font-size: 20px;
    padding-top: 0;
  }

  .search,
  .tryit, 
  .success {
    margin-top: 10px;
  }

  .instruction img {
    width: 30px; 
  }

  .artist-card {
    p {
      font-size: 20px;
      padding: 6px;
      margin: 0;
    }

    img {
      width: 35px;
      height: 35px;
    }
  }

  .tryit,
  .success {
    .artist-image {
      width: 150px;
      height: 150px;
    }

    ul {
      padding-left: 20px;
    }
  }
}

@media only screen and (max-width: 350px) {
  body button {
    font-size: 16px;
  }
} 
