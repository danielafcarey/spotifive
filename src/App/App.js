import React, { Component } from 'react';
import { Route } from 'react-router-dom';
import './App.css';

import Header from '../Header/Header';
import Search from '../Search/Search';
import Home from '../Home/Home';
import Tryit from '../Tryit/Tryit';
import Authorize from '../Authorize/Authorize';


class App extends Component {
  render() {
    return (
      <div className="App">
        <Header />
        <Route path='/authorize'
          component={ Authorize }
        />
        <Route 
          exact path='/'
          component={ Home }
        />
        <Route 
          exact path='/search'
          component={ Search }
        />
        <Route 
          exact path='/tryit'
          component={ Tryit }
        />
      </div>
    );
  }
}

export default App;
