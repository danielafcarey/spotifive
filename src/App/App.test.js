import React from 'react';
import { shallow } from 'enzyme';
import ReactDOM from 'react-dom';
import App from './App';

describe('App', () => {

  it('matches the snapshot', () => {

  })

  it('renders Signin if there is no user signed in', () => {

  })

  it('renders Search if there is a user signed in', () => {

  })

  it('renders Tryit if there is current artist info', () => {

  })

  describe('mapStateToProps', () => {

    it('returns a prop object with currentUser key and value', () => {

    })

    it('returns a prop object with currentArtist key and value', () => {

    })

  })

})
