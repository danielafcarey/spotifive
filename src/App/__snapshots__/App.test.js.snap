// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <header>
          <h1>
                    Spotifive
          </h1>
</header>,
        <Route
          component={[Function]}
          path="/authorize"
/>,
        <Route
          component={[Function]}
          exact={true}
          path="/"
/>,
        <Route
          component={[Function]}
          exact={true}
          path="/search"
/>,
        <Route
          component={[Function]}
          exact={true}
          path="/tryit"
/>,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <h1>
            Spotifive
</h1>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Spotifive",
          },
          "ref": null,
          "rendered": "Spotifive",
          "type": "h1",
        },
        "type": "header",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "path": "/authorize",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "exact": true,
          "path": "/",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "exact": true,
          "path": "/search",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "exact": true,
          "path": "/tryit",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <header>
            <h1>
                        Spotifive
            </h1>
</header>,
          <Route
            component={[Function]}
            path="/authorize"
/>,
          <Route
            component={[Function]}
            exact={true}
            path="/"
/>,
          <Route
            component={[Function]}
            exact={true}
            path="/search"
/>,
          <Route
            component={[Function]}
            exact={true}
            path="/tryit"
/>,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h1>
              Spotifive
</h1>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Spotifive",
            },
            "ref": null,
            "rendered": "Spotifive",
            "type": "h1",
          },
          "type": "header",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "path": "/authorize",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/search",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/tryit",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
